.env.sample

```diff
@@ -69,24 +69,33 @@ SLACK_SECRET=get_the_secret_of_above_key
 GOOGLE_CLIENT_ID=
 GOOGLE_CLIENT_SECRET=
 
-# To configure Microsoft/Azure auth, you'll need to create an OAuth Client. See 
+# To configure Microsoft/Azure auth, you'll need to create an OAuth Client. See
 # the guide for details on setting up your Azure App:
 # => https://wiki.generaloutline.com/share/dfa77e56-d4d2-4b51-8ff8-84ea6608faa4
-AZURE_CLIENT_ID= 
+AZURE_CLIENT_ID=
 AZURE_CLIENT_SECRET=
 AZURE_RESOURCE_APP_ID=
 
+# To configure generic OIDC auth, you'll need some kind of identity provider.
+# See documentation for whichever IdP you use to acquire the following info:
+# Redirect URI is https://<URL>/auth/oidc.callback
```

Comment:
```suggestion
# Redirect URI is https://<URL>/auth/oauth2.callback
```

Judging by the provider name this should be changed, I think that leaving this as `oidc` might be a better choice though and change the provider name, thoughts?

Reply:
@tommoor I'm inclined to agree... the passport strategy is a bit of a misnomer given the `userinfo` extension, which is part of the `oidc` spec 🤷

Either way, I agree it needs to be consistent. Good catch.

Reply:
Thoughts on using `openid-client`’s strategy rather than `passport-oauth2` which is rather outdated? I have an implementation I can share should you need, and I feel it’d be safer relying on OIDC-spec compliant libs

Reply:
Plus using OIDC means being able to utilize super kEwL features like PKCE/client-credentials flow (which means you literally don’t even have to supply OIDC client secret!!)

Reply:
Plus there’s really no “standard” way of getting user info in OAuth2 - it all depends on the identity provider’s implementation (so for facebook/github/twitter/etc you’d all need different ways of getting the user, *plus* there’s no “standardized” way of returning users, whereas with OIDC there is a strict, documented standard) - e.g. see `passport-facebook`, `passport-twitter`, `passport-$literallyHalfTheAuth` which have to extend `passport-oauth2` with bespoke “get user info” code

Reply:
@JaneJeon I actually started with `openid-client` before moving to `passport-oauth2` because I was blocked due to issues arising. See #2369 for the discussion, including some discussion of PKCE. In general `oidc-client` is the more popular and better maintained library. Perhaps we can refactor to it once the standing issues have been dealt with.

Regarding "standard" userinfo; the way I have done it in this PR is the standard way according to the OIDC spec, which is to use the supplied userinfo endpoint, and the acquired access token to resolve a user info document containing claims.

Additionally, even if we did have PKCE, the authorization code flow is still superior to the implicit flow from a security perspective, however that does not mean that authorization code flow does not benefit from PKCE.